js/reference.js

```diff
@@ -1162,7 +1164,8 @@
 
                         // column is not sortable
                         if (typeof sortCols === 'undefined') {
-                            throw new module.BadRequestError("", "Column " + sortObject[i].column + " is not sortable.");
+                            var path = parts = this.uri.match(/(.*)\/catalog\/([^\/]*)\/(entity|attribute|aggregate|attributegroup)\/(.*)/)[4];
```

Comment:
`this.location` returns the location object here, and then you have access to path (`this.location.path`).